stream-chat-android-client/src/main/java/io/getstream/chat/android/client/parser/ChatParser.kt

```diff
@@ -101,4 +105,25 @@ internal interface ChatParser {
             "\nMore information available at $moreInfo"
         } else ""
     }
+
+    private fun detailsTemplate(details: List<ErrorDetail>): String {
+        return if (details.isNotEmpty()) {
+            "\nError details: $details"
+        } else ""
+    }
+
+    private fun ErrorResponse.extractCause(): Throwable? {
+        if (code == ChatErrorCode.MESSAGE_MODERATION_FAILED.code) {
+            return MessageModerationFailedException(
+                details = details.map { detail ->
+                    MessageModerationFailedException.Detail(
```

Comment:
Do we have to map each Detail object into a Detail object that has the same properties? can't we just send the same list to the exception? We're mapping the same data to the same data here

Reply:
We're mapping `io.getstream.chat.android.client.socket.ErrorDetail` which is located inside `ErrorResponse` into `io.getstream.chat.android.client.errors.cause.MessageModerationFailedException.Detail`.

I didn't want to expose the socket/transport errors directly + I'm not sure if we always gonna map this way.